lib/index.js

```diff
@@ -74,21 +76,38 @@ class EvervaultClient {
     const isIgnoreRequest = (domain) =>
       this._isIgnoreRequest(domain, ignoreExact, ignoreEndsWith);
 
-    const initializationPromise = this.http.getCert().then((pem) => {
+    let x509 = null;
+
+    async function updateCertificate() {
+      const pem = await this.http.getCert();
+      let cert = pem.toString();
+      x509 = new X509Certificate(cert);
       tls.createSecureContext = (options) => {
         const context = origCreateSecureContext(options);
         context.context.addCACert(pem);
         return context;
       };
-    });
+    }
+
+    function isCertificateInvalid() {
+      if (x509 == null) {
+        return true;
+      }
+      let ts = Date.now();
+      let date = new Date(ts);
+      return (
+        date.getUTCDate() > x509.validTo || date.getUTCDate() < x509.validFrom
+      );
```

Comment:
I'm not sure this will work, `validTo` and `validFrom` give back date/time strings, so we probably want something like:
```suggestion
      const epoch = new Date().valueOf();
      return (
        epoch > (new Date(x509.validTo)).valueOf() || epoch < (new Date(x509.validFrom)).valueOf()
      );
```